docs/notebook_preamble.py

```diff
@@ -0,0 +1,68 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
```

Comment:
Can we add a brief comment here what the purpose of this file is, and how/where it gets included in the docs? I know this is obvious in this PR, but if someone looks at it separately it may not be clear why it contains various disparate functions and how it gets integrated with the notebook(s) (i.e. via a setting in `mkdocs.yml`).